tools/isobuild/compiler-plugin.js

```diff
@@ -1720,21 +1720,21 @@ export class PackageSourceBatch {
     const fileHashes = [];
     const cacheKeyPrefix = sha1(JSON.stringify({
       linkerOptions,
-      files: await jsResources.reduce(async (acc, inputFile) => {
-        const resolvedAcc = await acc;
-
-        fileHashes.push(await inputFile.hash);
-        return [...resolvedAcc, {
-          meteorInstallOptions: inputFile.meteorInstallOptions,
-          absModuleId: inputFile.absModuleId,
-          sourceMap: !! await inputFile.sourceMap,
-          mainModule: inputFile.mainModule,
-          imported: inputFile.imported,
-          alias: inputFile.alias,
-          lazy: inputFile.lazy,
-          bare: inputFile.bare,
-        }];
-      }, Promise.resolve([]))
+      files: await Promise.all(
+        jsResources.map(async (inputFile) => {
+          fileHashes.push(await inputFile.hash);
+          return {
+            meteorInstallOptions: inputFile.meteorInstallOptions,
+            absModuleId: inputFile.absModuleId,
+            sourceMap: !!(await inputFile.sourceMap),
+            mainModule: inputFile.mainModule,
+            imported: inputFile.imported,
+            alias: inputFile.alias,
+            lazy: inputFile.lazy,
+            bare: inputFile.bare,
+          };
+        })
+      )
```

Comment:
This is actually the pattern that enables parallelization on parsing the code from files and speeds up the build.